ingenialink/canopen/network.py

```diff
@@ -239,6 +244,42 @@ def scan_slaves(self) -> List[int]:
 
         return nodes  # type: ignore [no-any-return]
 
+    def scan_slaves_info(self) -> OrderedDict[int, SlaveInfo]:
+        """Scans for nodes in the network and return an ordered dict with the slave information.
+
+        Returns:
+            Ordered dict with the slave information.
+
+        """
+        try:
+            slaves = self.scan_slaves()
+        except ILError:
+            return OrderedDict()
```

Comment:
Why did you use OrderedDict instead of a regular dict?

Reply:
To ensure that the drives are correctly ordered in the Dict. So we could potentially use the keys of the dictionary as a list of scanned slaves.